cfgmassif.go

```diff
@@ -24,19 +26,57 @@ func cfgMassifReader(cmd *CmdCtx, cCtx *cli.Context) error {
 		}
 	}
 
-	if cmd.reader == nil {
-		if err = cfgReader(cmd, cCtx); err != nil {
-			return err
-		}
-	}
-
-	massifReader := massifs.NewMassifReader(logger.Sugar, cmd.reader, massifs.WithoutGetRootSupport())
-	cmd.massifReader = massifReader
 	cmd.massifHeight = uint8(cCtx.Uint("height"))
 	if cmd.massifHeight == 0 {
 		cmd.massifHeight = defaultMassifHeight
 	}
 
+	localLog := cCtx.String("data-local")
+	remoteLog := cCtx.String("data-url")
+
+	if localLog != "" && remoteLog != "" {
+		return fmt.Errorf("can't use data-local and data-url at the same time")
+	}
+
+	if localLog == "" && remoteLog == "" {
+		// if we had no url and no local data supplied we try STDIN
+		opener := cfgStdinOpener()
+		mr, err := NewLocalMassifReader(logger.Sugar, opener, "-")
+		if err != nil {
+			return err
+		}
+		cmd.massifReader = mr
+	} else if localLog != "" {
+		// if we have local log then we try to figure out if it's a dir or
+		// a file and we go with that
+		absLocal, err := filepath.Abs(localLog)
+		if err != nil {
+			return err
+		}
+		fi, err := os.Stat(absLocal)
+		if err != nil {
+			return err
+		}
+		opts := []Option{}
+		if fi.IsDir() {
```

Comment:
Potentially, we can fail here if its not a directory

Reply:
we also accept a single file